core/src/main/java/org/elasticsearch/search/aggregations/InternalAggregation.java

```diff
@@ -149,9 +149,19 @@ protected static void writeSize(int size, StreamOutput out) throws IOException {
         return pipelineAggregators;
     }
 
+    /**
+     * Returns the name of the aggregation preceded by a prefix that uniquely
+     * identifies its type. This prefix can later be used by REST clients to
+     * determine the internal type of the aggregation.
+     */
+    protected String getPrefixedName() {
```

Comment:
I would rather define a protected method that only returns the prefix (`getType` ?), whose default return value is `getWriteableName()`. Otherwise subclasses can potentially override the separator used (or even whether they include the name of the agg or not).